.golangci.yml

```diff
@@ -0,0 +1,116 @@
+output:
+  sort-results: true
+
+issues:
+  exclude-use-default: false
+  max-same-issues: 0
+  max-issues-per-linter: 0
+  exclude-rules:
+    # False positive: https://github.com/kunwardeep/paralleltest/issues/8.
+    - linters:
+        - paralleltest
+      text: "does not use range value in test Run"
+
+linters-settings:
+  errcheck:
+    check-type-assertions: true
+    check-blank: true
+  gci:
+    local-prefixes: github.com/flatcar-linux/fleetlock
+  godot:
+    capital: true
+  gofumpt:
+    extra-rules: true
+  govet:
+    enable-all: true
+    disable:
+      - fieldalignment
+      - shadow
+  makezero:
+    always: true
+  nolintlint:
+    allow-leading-space: false
+    require-explanation: true
+    require-specific: true
+  wsl:
+    force-err-cuddling: true
+
+linters:
+  disable-all: false
+  disable:
+    # We do not have clearly defined error types yet.
```

Comment:
What about opening an issue to track this ? Can be a "good first issue" I guess.

Reply:
I'm not sure if this is really worth the effort. So right now, when you enable the linter, it gives the following:
```console
$ golangci-lint run
WARN [runner] The linter 'scopelint' is deprecated (since v1.39.0) due to: The repository of the linter has been deprecated by the owner.  Replaced by exportloopref.
pkg/client/client.go:146:10: err113: do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"fleetlock error: %s\", e.String())" (goerr113)
                return fmt.Errorf("fleetlock error: %s", e.String())
                       ^
pkg/client/client.go:148:10: err113: do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"unexpected status code: %d\", resp.StatusCode)" (goerr113)
                return fmt.Errorf("unexpected status code: %d", resp.StatusCode)
                       ^
pkg/client/client_test.go:57:16: err113: do not define dynamic errors, use wrapped static errors instead: "errors.New(\"fleetlock error: this is an error (error_kind)\")" (goerr113)
                        expErr:     errors.New("fleetlock error: this is an error (error_kind)"),
                                    ^
pkg/client/client_test.go:62:16: err113: do not define dynamic errors, use wrapped static errors instead: "errors.New(\"unmarshalling error: invalid character '\\\"' after object key:value pair\")" (goerr113)
                        expErr:     errors.New("unmarshalling error: invalid character '\"' after object key:value pair"),
                                    ^
pkg/client/client_test.go:67:16: err113: do not define dynamic errors, use wrapped static errors instead: "errors.New(\"unexpected status code: 100\")" (goerr113)
                        expErr:     errors.New("unexpected status code: 100"),
                                    ^
pkg/client/client_test.go:70:12: err113: do not define dynamic errors, use wrapped static errors instead: "errors.New(\"doing the request: connection refused\")" (goerr113)
                        expErr: errors.New("doing the request: connection refused"),
                                ^
pkg/client/client_test.go:71:12: err113: do not define dynamic errors, use wrapped static errors instead: "errors.New(\"connection refused\")" (goerr113)
                        doErr:  errors.New("connection refused"),
                                ^

```

So roughly, we would either have to have one type catch all type of error or be more specific and have like connection error, FleetLock error, configuration error etc.

Situations when you need to actually differentiate between this kind of errors and handle them differently are rare from my experience. 

>What about opening an issue to track this ? Can be a "good first issue" I guess.

I guess that place would be https://github.com/flatcar-linux/flatcar? That seems a bit too specific for place like this to me :smile:

Reply:
> I guess that place would be https://github.com/flatcar-linux/flatcar? That seems a bit too specific for place like this to me

yes definitely, let's open the "issue" tab.